### PR TITLE
Feature/upgrade services

### PR DESCRIPTION
**Pull request by rahushen**
- Added new property 'package'
- Multiple URIs now supported (bootflash:, usb, tftp:)
- Deprecated 'version' property

Relevant beaker tests ran and passed.

### CODE REVIEW

File: examples/cisco/demo_upgrade.pp [hunk 1]
```diff
@@ -69,7 +68,7 @@
     }
 
     cisco_upgrade { 'image' :
-      version           => "${version}",
+      package           => "${image}",
```
Comment (mikewiebe) [inline examples/cisco/demo_upgrade.pp]: As we discussed combine `package` and `source_uri` into a single property that can manage the device.

The output of the `puppet resource` command show display `package` information and `version` information.

**Comment by mikewiebe** [comment]
@rahushen Let me know when this is ready to review.  I see changes going in but will wait until it's complete.

File: lib/puppet/type/cisco_upgrade.rb [hunk 2]
```diff
@@ -101,16 +75,33 @@ def self.title_patterns
...
+    desc "{ackage to install on the device. Format <uri>:<image>.
+          Valid values are string.#{examples}#{supported}"
+    validate do |pkg|
+      fail 'Package should be a string.' unless pkg.is_a?(String)
+      fail 'package must match format <uri>:<image>' unless pkg[/\S+:\S+/]
+    end
+  end # property package
+
+  validate do
+    fail "The property 'version' has been deprecated. Please use 'package'." if
```
Comment (mikewiebe) [inline lib/puppet/type/cisco_upgrade.rb]: Should you also check for `source_uri` here as a deprecated parameter?

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,9 +13,17 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
 
+- Added support for tftp and usb URIs to `cisco_upgrade`
+
+- Extend `cisco_upgrade` with attributes:
+ - `package`
+
 ### Changed
 
 ### Removed
+- Deprecated attributes for `cisco_upgrade`:
```
Comment (mikewiebe) [inline CHANGELOG.md]: I would also indicate that these properties are deprecated in favor of a single propertly `package` to manage the resource.

File: README.md [hunk 4]
```diff
@@ -4075,8 +4070,10 @@ Force upgrade the device.Valid values are `true`, `false`.
 
 #### Properties
 
-##### `version`
-Version of the Cisco image to install on the device. Valid values are strings.
+##### `package`
+Package to install on the device. Format `<uri>:<image>`. Valid values are strings.
+*Example --> bootflash:nxos.7.0.3.I5.2.bin*
```
Comment (mikewiebe) [inline README.md]: I suggest you add an example using tftp as well.

File: lib/puppet/type/cisco_upgrade.rb (hunk 2 above)
Comment (rahushen) [inline lib/puppet/type/cisco_upgrade.rb]: I have taken out the source_uri parameter. Manifest will fail if 'source_uri' is used.

File: README.md (hunk 4 above)
Comment (rahushen) [inline README.md]: Done.

File: CHANGELOG.md (hunk 3 above)
Comment (rahushen) [inline CHANGELOG.md]: Done.

File: lib/puppet/type/cisco_upgrade.rb (hunk 2 above)
Comment (mikewiebe) [inline lib/puppet/type/cisco_upgrade.rb]: 👍

**Pull request merged**